Plots/Contours/NCL_eof_1_1.py

```diff
@@ -0,0 +1,326 @@
+
+###############################################################################
+# Compute desired global seasonal mean using month_to_season()
+
+# Choose the winter season (December-January-February)
+season = "DJF"
+SLP = month_to_season(ds, season)
+print_debug('\n\nSLP:\n\n')
+print_debug(SLP)
+
+# Diagnostic plot: show slice of SLP
+sliceSLP = SLP.sel(lat=slice(latS, latN), lon=slice(lonL, lonR))
+
+print_debug('\n\nsliceSLP:\n')
+print_debug(sliceSLP)
+
+###############################################################################
+# Create weights: sqrt(cos(lat))   [or sqrt(gw) ]
+
+deg2rad = 4. * atan(1.) / 180.
+clat = SLP['lat'].astype(dtype=np.float64)
+clat = np.sqrt(np.cos(deg2rad * clat))
+print_debug('\n\nclat:\n')
+print_debug(clat)
+
+###############################################################################
+# Multiply SLP by weights:
+
+# Xarray uses the supplied coordinate information to apply latitude-based
+# weights to all longitudes and timesteps automatically.
+
+wSLP = SLP
+wSLP['slp'] = clat * SLP['slp']
+
+# For now, metadata for slp must be copied over explicitly; it is not preserved by binary operators like multiplication.
+wSLP['slp'].attrs = ds['slp'].attrs
+wSLP['slp'].attrs['long_name'] = 'Wgt: ' + wSLP['slp'].attrs['long_name']
+
+###############################################################################
+# Subset data to the North Atlantic region:
+
+xw = wSLP.sel(lat=slice(latS, latN), lon=slice(lonL, lonR))
+
+print_debug('\n\nxw:\n\n')
+print_debug(xw.slp)
+
+###############################################################################
+# Compute the EOFs:
+
+eof = eofunc(xw["slp"], neof, time_dim=1, meta=True)
+
+print_debug('\n\neof:\n\n')
+print_debug(eof)
+
+eof_ts = eofunc_ts(xw["slp"], eof, time_dim=1, meta=True)
+
+print_debug('\n\neof_ts:\n\n')
+print_debug(eof_ts)
+
+###############################################################################
+# Normalize time series:
+
+# Sum spatial weights over the area used.
+nLon = len(xw['lon'])
+clat_subset = clat.where((clat.lat >= latS) & (clat.lat <= latN), drop=True)
+weightTotal = clat_subset.sum() * nLon
+eof_ts = eof_ts / weightTotal
+
+print_debug('\n\neof_ts normalized:\n\n')
+print_debug(eof_ts)
+
+###############################################################################
+# Utility function:
+
+# Define a utility function for creating a contour plot.
+def make_contour_plot(ax, dataset):
+
+    lat = dataset['lat']
+    lon = dataset['lon']
+    values = dataset.data
+
+    # Import an NCL colormap
+    cmap = gvcmaps.BlWhRe
+
+    # Specify contour levelstamam
+    v = np.linspace(-0.08, 0.08, 9, endpoint=True)
+
+    # The function contourf() produces fill colors, and contour() calculates contour label locations.
+    cplot = ax.contourf(lon, lat, values, levels=v, cmap=cmap, extend="both", transform=ccrs.PlateCarree())
+    p = ax.contour(lon, lat, values, levels=v, linewidths=0.0, transform=ccrs.PlateCarree())
```

Comment:
`0.0` linewidth seems funny; can we do `0.1`?

Reply:
The original plot has no contour lines.   We need to call the function to calculate contour label locations only.   This is the only way I know to get them.

Reply:
oh hmm... add a comment then?

Reply:
actually can't you pass `cplot` to the `clabel` function?

Reply:
I tried doing this, and it did not behave well for me.    Strange triangular artifacts appear on the plot when I do this.    If you see a way around, or a different outcome than me, let me know.